src/v/cluster/tm_stm.cc

```diff
@@ -117,16 +117,77 @@ tm_stm::tm_stm(
       config::shard_local_cfg().tm_violation_recovery_policy.value())
   , _feature_table(feature_table) {}
 
-ss::future<std::optional<tm_transaction>>
-tm_stm::get_tx(const kafka::transactional_id& tx_id) {
+std::optional<tm_transaction>
+tm_stm::do_get_tx(const kafka::transactional_id& tx_id) {
```

Comment:
should this be folded into the previous commit?

Reply:
I think not, this is core of the change while last one is just cosmetic coroutinization.